app/controllers/annotations_controller.rb

```diff
@@ -119,4 +49,28 @@ def init_annotation(text, question, org, type)
     return annotation
   end
 
+  def process_changes(hash, input, type)
+    # If the input is available update the annotation otherwise remove it if it exists
+    if input.present? 
+      annotation = Annotation.find_or_create_by(hash)
```

Comment:
find_or_create does not guarantee exclusivity if two or more threads are running same line of code, in case you wanted to avoid race conditions per database connection consider using transactions.